### PR TITLE
Add npo:recents extractor

### PR DESCRIPTION
**Pull request by jhoek**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extractor for npo.nl _programs_ (as opposed to _episodes_ of programs). Retrieves only the most recent episodes of the program in question (hence the name). Some programs have so many episodes available that it doesn't make any practical sense to retrieve all, as discussed in issue #7947. I am not a Python developer (as you may have noticed...), and I wasn't confident enough about my code to incorporate it into npo.py, where it probably belongs?!

### CODE REVIEW

File: youtube_dl/extractor/nporecents.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            'description': 'In dit programma staat centraal wat fabrikanten ons als consumenten vertellen. Klopt het wat ze claimen en wat ze ons in reclames verkopen? Verslaggevers Teun van de Keuken, Sofie van den Enk, Daan Nieber, Ersin Kiris, Marijn Frank en Maarten Remmers nemen de telefoon ter hand en bellen er actief op los. Ze stellen simpele vragen en krijgen de meest verbazingwekkende antwoorden op food, non-food en nieuwsgerelateerde kwesties. Prikkelend, onderzoekend en vasthoudend. Keuringsdienst van Waarde: simpele vragen,verbazingwekkende antwoorden.'
+        },
+        'playlist_mincount': 8
+    }
+
+    def _extract_entries(self, webpage, program_id, program_url):
+        is_npo3 = 'www-assets.npo.nl/uploads/tv_channel/265/logo/smaller_npo3-logo.png' in webpage
+
+        if is_npo3:
+            episodes_url = '{}//search?category=broadcasts&page=1'.format(
```
Comment (dstftw) [inline youtube_dl/extractor/nporecents.py]: `{}` does not work in python 2.6.

File: youtube_dl/extractor/nporecents.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    yield self.url_result(
+                        url='http://npo.nl{}'.format(hyperlink.get('href')),
+                        video_title=self._og_search_title(webpage))
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        alt_id = mobj.group('alt_id')
+        program_id = mobj.group('program_id')
+        webpage = self._download_webpage(url, program_id)
+        title = self._og_search_title(webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/nporecents.py]: Should be non fatal.

File: youtube_dl/extractor/nporecents.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+class NPORecentsIE(InfoExtractor):
+    IE_Name = 'npo:recents'
+    _VALID_URL = r'(?:https?://)?(?:www\.)?npo\.nl/(?P<alt_id>[^/]+)/(?P<program_id>\w+_\d+)'
+    _TEST = {
+        'url': 'https://www.npo.nl/keuringsdienst-van-waarde/KN_1678993',
+        'info_dict': {
+            'title': 'Keuringsdienst van Waarde',
+            'id': 'KN_1678993',
+            'description': 'In dit programma staat centraal wat fabrikanten ons als consumenten vertellen. Klopt het wat ze claimen en wat ze ons in reclames verkopen? Verslaggevers Teun van de Keuken, Sofie van den Enk, Daan Nieber, Ersin Kiris, Marijn Frank en Maarten Remmers nemen de telefoon ter hand en bellen er actief op los. Ze stellen simpele vragen en krijgen de meest verbazingwekkende antwoorden op food, non-food en nieuwsgerelateerde kwesties. Prikkelend, onderzoekend en vasthoudend. Keuringsdienst van Waarde: simpele vragen,verbazingwekkende antwoorden.'
```
Comment (dstftw) [inline youtube_dl/extractor/nporecents.py]: Replace with `'md5:...'`.

File: youtube_dl/extractor/nporecents.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        if is_npo3:
+            episodes_url = '{}//search?category=broadcasts&page=1'.format(
+                program_url)
+        else:
+            episodes_url = '{}/search?media_type=broadcast&start=0&rows=8'.format(
+                program_url)
+
+        episodes = self._download_webpage(
+            episodes_url, program_id, note='Retrieving episodes')
+        tree = ET.fromstring(episodes.encode('utf-8'))
```
Comment (dstftw) [inline youtube_dl/extractor/nporecents.py]: Do not parse webpage as XML.

**Comment by dstftw** [comment]
1. Read
>[adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
2. Test your code under python 2.6+ and python 3.2+.
3. Check your code with flake8.
4. Move extractor to `npo.py`.

File: youtube_dl/extractor/nporecents.py (hunk 3 above)
Comment (jhoek) [inline youtube_dl/extractor/nporecents.py]: Done.

File: youtube_dl/extractor/nporecents.py (hunk 1 above)
Comment (jhoek) [inline youtube_dl/extractor/nporecents.py]: Done.

File: youtube_dl/extractor/nporecents.py (hunk 4 above)
Comment (jhoek) [inline youtube_dl/extractor/nporecents.py]: Done; gone back to regex-based approach.

File: youtube_dl/extractor/nporecents.py (hunk 2 above)
Comment (jhoek) [inline youtube_dl/extractor/nporecents.py]: Done.

**Comment by jhoek** [comment]
Travis still seems to "see" the nporecents.py, despite that file being removed in one of the commits, which leads to duplicate extractors. 

Never mind, I'll close this PR and try again some time later.

**Pull request closed without merging**